t/ui/16-tests_dependencies.t

```diff
@@ -148,6 +129,54 @@ subtest 'dependency json' => sub {
     diag explain $t->tx->res->json unless $t->success;
 };
 
+subtest 'dependency JSON after duplicating jobs' => sub {
+    # remove the job 99938 is cloned as to make this test more distinct from the previous subtest
+    $schema->txn_begin;
+    $textmode_job->delete;
+
+    # duplicate one of the jobs in the parallel cluster (which will also duplicate a few dependend jobs)
+    my $duplicates = $jobs->find(99963)->auto_duplicate->{cluster_cloned};
+    my @new_jobs = sort values %$duplicates;
+    my $directly_chained_child_dup = $duplicates->{99927};
+    is_deeply [sort keys %$duplicates], [99927, 99961, 99963], 'expected set of jobs has been duplicated';
+
+    subtest 'graph for the original jobs' => sub {
+        my $json = $t->get_ok('/tests/99963/dependencies_ajax')->status_is(200)->tx->res->json;
+        my @rendered_nodes = sort map { $_->{id} } @{$json->{nodes}};
+        my @expected_nodes = (
+            99927,    # because it has the same clone-depth as the original job
+            99937,    # because it has not been duplicated
+            99938,    # because it has not been duplicated
+            99961,    # because it has the same clone-depth as the original job
+            99963,    # because it is the original job
+            @new_jobs    # because those are indirect dependencies
+        );
+        is_deeply \@rendered_nodes, \@expected_nodes, 'all original jobs are present as well'
+          or diag explain \@rendered_nodes, $duplicates;
+
+        $json = $t->get_ok('/tests/99961/dependencies_ajax')->status_is(200)->tx->res->json;
+        @rendered_nodes = sort map { $_->{id} } @{$json->{nodes}};
+        @expected_nodes = grep { $_ != $directly_chained_child_dup } @expected_nodes;
+        is_deeply \@rendered_nodes, \@expected_nodes,
+          'same outcome for parallel sibling except that directly chained child is not indirectly added'
+          or diag explain \@rendered_nodes, $duplicates;
+    };
+
+    subtest 'graph for the duplicated jobs' => sub {
+        my $json = $t->get_ok("/tests/$duplicates->{99963}/dependencies_ajax")->status_is(200)->tx->res->json;
+        my @rendered_nodes = sort map { $_->{id} } @{$json->{nodes}};
+        my @expected_nodes = (
+            99937,    # because it has not been duplicated
+            99938,    # because it has not been duplicated
+            @new_jobs    # because those are the duplicated jobs
+        );
+        is_deeply \@rendered_nodes, \@expected_nodes, 'only the latest jobs are shown'
+          or diag explain \@rendered_nodes;
+    };
+
+    $schema->txn_rollback;
```

Comment:
> I have tested it manually and it works but I'll keep it a draft because I need to extend unit tests.

I think the description of the PR is lying 😉

Reply:
solved